app/controllers/report_controller/widgets.rb

```diff
@@ -122,12 +122,12 @@ def widget_edit
   def widget_delete
     assert_privileges("widget_delete")
     widgets = find_checked_items
-    if !params[:id].nil? && MiqWidget.find_by_id(params[:id]).nil?
+    if params[:id].present? && !MiqWidget.exists?(params[:id])
```

Comment:
This whole logic of finding the record seems twisted. I believe, we have a better parent and methods for that. Maybe @romanblanco or @PanSpagetka can fix this in a separate PR?